src/datasets/features/audio.py

```diff
@@ -150,20 +148,49 @@ def decode_example(
         path, file = (value["path"], BytesIO(value["bytes"])) if value["bytes"] is not None else (value["path"], None)
         if path is None and file is None:
             raise ValueError(f"An audio sample should have one of 'path' or 'bytes' but both are None in {value}.")
-        elif path is not None and path.endswith("mp3"):
-            array, sampling_rate = self._decode_mp3(file if file else path)
-        elif path is not None and path.endswith("opus"):
-            if file:
-                array, sampling_rate = self._decode_non_mp3_file_like(file, "opus")
-            else:
-                array, sampling_rate = self._decode_non_mp3_path_like(
-                    path, "opus", token_per_repo_id=token_per_repo_id
+
+        try:
+            import librosa
+            import soundfile as sf
+        except ImportError as err:
+            raise ImportError("To support decoding audio files, please install 'librosa' and 'soundfile'.") from err
+
+        audio_format = os.path.splitext(path)[1].lstrip(".").lower() if path is not None else None
```

Comment:
I think we need to use `xsplitext` (from `download/streaming_download_manager.py`) here to account for URLs

```suggestion
        audio_format = xsplitext(path)[1][1:].lower() if path is not None else None
```